### PR TITLE
New interface for validation splits

### PR DESCRIPTION
**Pull request by hfrick**
todo before publication
- [x] update date
- [x] re-render without dev versions
- [x] add link to release notes

Rendered post: https://deploy-preview-653--tidyverse-org.netlify.app/blog/2023/08/validation-split-as-3-way-split/

### CODE REVIEW

File: content/blog/validation-split-as-3-way-split/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+output: hugodown::hugo_document
+
+slug: validation-split-as-3-way-split
+title: New interface to validation splits
```
Comment (hfrick) [inline content/blog/validation-split-as-3-way-split/index.Rmd:5]: We've talked about it as a new interface but I'm not sure this makes for the best title. Should this be something more general / less jargon-y like "A new way to make a validation split"? (or set?)

File: content/blog/validation-split-as-3-way-split/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+<!-- But `tune_grid()` also needs data to fit the model on and data to assess the fitted model on! That's the training set and the validation set. The test set needs to be put aside until we've picked our final model. -->
+<!-- The new `validation_set()` function takes our initial three-way split object and returns the training set and the validation set, ready for use with the tuning functions like `tune_grid()`. -->
+
+During tuning, the model should not have access to the test data, only to the data used to fit the model (the analysis set) and the data used to assess the model (the assessment set). Each pair of analysis and assessment set forms a resample. With a validation split, we have just one resample with the training set functioning as the analysis set and the validation set as the assessment set. For 10-fold cross-validation, we'd have 10 resamples. The tidymodels tuning functions all expect a _set_ of resamples and the corresponding objects are of class `rset`. 
+To remove the test data from the initial three-way split and create such an `rset` object for tuning, use `validation_set()`.
```
Comment (hfrick) [inline content/blog/validation-split-as-3-way-split/index.Rmd:121]: I've turned this over a few times and would love feedback on whether this is a clear motivation for the function name that also works for someone who is not too deeply immersed in `rsplit` vs `rset`.

Comment (EmilHvitfeldt) [reply content/blog/validation-split-as-3-way-split/index.Rmd:121]: I think the sentence "For 10-fold cross-validation, we'd have 10 resamples." doesn't fit well. Maybe move it forward so this paragraph goes

- In tune we need analysis and assessment
- a pair of analysis and assessment forms a resample
- in 10-fold CV we have 10 resamples
- When using validation splits, training = analysis, validation = assessment
- thus validation splits gives us 1 resample

File: content/blog/validation-split-as-3-way-split/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+`last_fit()` then lets you fit your model on the training data and calculate performance on the test data. If you provide it with a three-way split, you can choose if you want your model to be fitted on the training data only or on the combination of training and validation set. You can specify this with the `add_validation_set` argument.
+
+```{r last-fit}
+childcare_fit <- last_fit(childcare_wflow, childcare_split, add_validation_set = TRUE)
+collect_metrics(childcare_fit)
+```
+
+## Acknowledgements
```
Comment (EmilHvitfeldt) [inline content/blog/validation-split-as-3-way-split/index.Rmd:145]: Do we needed a ending sentence here to sum up what we did?

**Review by EmilHvitfeldt — APPROVED** [review]
Few notes:

File: content/blog/validation-split-as-3-way-split/index.Rmd (hunk 2 above)
Comment (hfrick) [reply content/blog/validation-split-as-3-way-split/index.Rmd:121]: Thanks! I think that works better 👍
